src/libraries/Microsoft.PowerFx.Connectors/OpenApiParser.cs

```diff
@@ -0,0 +1,138 @@
+
+                    if (op.RequestBody != null)
+                    {
+                        // RequestBody can be ambiguous- treat as a single record? splat as parameters?
+                        throw new NotImplementedException($"Request body");
+                    }
+
+                    var returnType = op.GetReturnType();
+
+                    if (basePath != null)
+                    {
+                        path = basePath + path;
+                    }
+
+                    var argMapper = new ArgumentMapper(op.Parameters);
+
+                    IAsyncTexlFunction invoker = null;
+                    if (httpClient != null)
+                    {
+                        var httpInvoker = new HttpFunctionInvoker(httpClient, verb, path, returnType, argMapper, cache);
+                        invoker = new ScopedHttpFunctionInvoker(functionNamespace, httpInvoker);
+                    }
+
+                    // Parameter (name,type) --> list of options. 
+                    var parameterOptions = new Dictionary<TypedName, List<string>>();
+                    var parameterDefaultValues = new Dictionary<string, Tuple<string, DType>>(StringComparer.Ordinal);
+
+                    var isBehavior = !IsSafeHttpMethod(verb);
+                    var isDynamic = false;
+                    var isAutoRefreshable = false;
+
+                    var isCacheEnabled = false;
+                    var cacheTimeoutMs = 10000;
+                    var isHidden = false;
+
+                    var description = op.Description ?? $"Invoke {operationName}";
+
+                    var sfunc = new ServiceFunction(
+                        null,
+                        theNamespace,
+                        operationName,
+                        operationName,
+                        description, // Template.GetFunctionDescription(funcTemplate.Name),
+                        returnType._type,
+                        BigInteger.Zero,
+                        argMapper._arityMin,
+                        argMapper._arityMax,
+                        isBehavior,
+                        isAutoRefreshable,
+                        isDynamic,
+                        isCacheEnabled,
+                        cacheTimeoutMs,
+                        isHidden,
+                        parameterOptions,
+                        argMapper._optionalParamInfo,
+                        argMapper._requiredParamInfo,
+                        parameterDefaultValues,
+                        "action", //  funcTemplate.ActionName,??
+                        argMapper._parameterTypes)
+                    {
+                        _invoker = invoker
+                    };
+
+                    newFunctions.Add(sfunc);
+                }
+            }
+
+            return newFunctions;
+        }
+
+        private static bool IsSafeHttpMethod(HttpMethod httpMethod)
```

Comment:
HttpMethod.Options is safe too.

Reply:
Just copying from Power Apps. 

Spec still only lists Get/Head
[https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html)

But Options does seem to be considered safe in practice:
[https://developer.mozilla.org/en-US/docs/Glossary/Safe/HTTP](https://developer.mozilla.org/en-US/docs/Glossary/Safe/HTTP)